src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,425 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import '@itwin/itwinui-css/css/date-picker.css';
+
+const isSameHour = (
+  date1: Date,
+  date2: Date | undefined,
+  period?: 'AM' | 'PM',
+) => {
+  const adjustedHours = period
+    ? formatHourFrom12(date1.getHours(), period)
+    : date1.getHours();
+  if (!!period) {
+    return !!date2 && adjustedHours % 12 === date2.getHours() % 12;
+  }
+  return !!date2 && adjustedHours === date2.getHours();
+};
+
+const areSameMinutes = (date1: Date, date2: Date | undefined) => {
```

Comment:
Maybe rename to `isSameMinute`, `isSameSecond` to be consistent also `are*` looks odd.

Reply:
renamed